server/src/test/java/org/elasticsearch/action/ActionListenerTests.java

```diff
@@ -434,13 +434,11 @@ public void testAssertAtLeastOnceWillNotLogWhenResolvedOrFailed() throws Excepti
                 listenerRef.get().onFailure(new RuntimeException("Failed"));
             }
         });
-        assertBusy(() -> {
-            // Nullify reference so it becomes unreachable
-            listenerRef.set(null);
-            System.gc();
-            reachabilityChecker.ensureUnreachable();    // Only proceed once we know the object isn't reachable
-            assertFalse(notCalledListenerCalled.get());
-        });
+        // Nullify reference so it becomes unreachable
+        listenerRef.set(null);
+        System.gc();
```

Comment:
I was thinking even this System.gc() is not needed.

Reply:
Good call, removed